src/transformStateWithClones.js

```diff
@@ -7,7 +7,50 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const TYPE_ONE = 'addProperties';
+  const TYPE_TWO = 'removeProperties';
+  const TYPE_THREE = 'clear';
+  const stateClone = { ...state };
+  const resultArr = [];
+
+  for (const obj of actions) {
+    const { type } = obj;
+
+    switch (type) {
+      case TYPE_ONE:
+        Object.assign(stateClone, obj.extraData);
+
+        const addObj = { ...stateClone };
+
+        resultArr.push(addObj);
```

Comment:
```suggestion

```
Now you are making the same thing 3 times in each case. Consider moving the push to the end of the loop and invoking it just once